nodejs/awsx/ec2/vpc.ts

```diff
@@ -124,8 +126,6 @@ export class Vpc extends pulumi.ComponentResource {
             throw new Error("Cannot add InternetGateway to Vpc that already has one.");
         }
 
-        opts = { parent: this, ...opts };
-
         // See https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Internet_Gateway.html#Add_IGW_Attach_Gateway
         // for more details.
         this.internetGateway = new x.ec2.InternetGateway(name, this, args, opts);
```

Comment:
Why is this forwarding `opts` instead of just parenting the resource?

Reply:
yeah, good point.  will fix.

Reply:
oh, nevermind. this is the instance method on vpc to make a natgateway.  We basically will parent with 'this vpc' by default, but we allow the caller to pass in their own opts in case they want to control this.  so ths is by design.  

The *direct* resources a component creates (i.e. in the constructor), are parented by teh component.  however, indirect resources (i.e. resources that can be created by a external party *from* the resource), will parent themselves by default, but will allow overriding by the external party.

Reply:
Note: this is also why we use the pattern `{ parent: x, ...opts }`.  That way, if `opts` includes a parent, it is respected. However, if it doesn't include one, we can pick a reasonable default.